lib/mix/lib/mix/generator.ex

```diff
@@ -115,7 +115,7 @@ defmodule Mix.Generator do
   to override it. Returns true otherwise.
   """
   @doc since: "1.9.0"
-  @spec overwrite?(Path.t(), iodata) :: boolean
+  @spec overwrite?(Path.t()) :: boolean
```

Comment:
I was planning on reporting an issue about this. But I wan to check if it's possible to solve it first.

Is there a way the compiler can detect this kind of error and warn about it at least? the dame way when we misplace function definitions that are in the wrong order

Reply:
Not really since someone can today just put all `@spec` at the top of the module. Plus, given that the `spec` repeats the name, it would be weird to enforce "sequence" AND expect the name. We could have made somethitng like:

    @spec arg1, arg2 -> return

But it is too late for that and if we are doing it we probably need a general overhaul of how specs work (which we don't plan in doing for now).

Reply:
Alright, I guess it could be a little job for a Credo or any other linter